src/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1460,6 +1563,17 @@ internal struct NativeVariant
 #endif
     }  // struct NativeVariant
 
+    // This NativeDecimal type is very similar to the System.Decimal type, except it requires an 8-byte alignment
+    // like the native DECIMAL type instead of the 4-byte requirement of the System.Decimal type.
```

Comment:
This requirement isn't really documented anywhere. It's just a fall-out of the various implementations. The native implementation has an 8-byte field overlapping the last two 4-byte integers, but (because of the old bad support for explicit layout) the managed implementation of Decimal has only 4-byte integer fields.